src/rgw/rgw_ssd_driver.cc

```diff
@@ -500,12 +501,19 @@ std::string SSDDriver::get_attr(const DoutPrefixProvider* dpp, const std::string
 
 int SSDDriver::set_attr(const DoutPrefixProvider* dpp, const std::string& key, const std::string& attr_name, const std::string& attr_val, optional_yield y)
 {
+    int e;
     std::string location = partition_info.location + key;
     ldpp_dout(dpp, 20) << "SSDCache: " << __func__ << "(): location=" << location << dendl;
 
     ldpp_dout(dpp, 20) << "SSDCache: " << __func__ << "(): set_attr: key: " << attr_name << " val: " << attr_val << dendl;
 
-    return setxattr(location.c_str(), attr_name.c_str(), attr_val.c_str(), attr_val.size(), 0);
+    int ret = setxattr(location.c_str(), (ATTR_PREFIX + attr_name).c_str(), attr_val.c_str(), attr_val.size(), 0);
```

Comment:
@samarahu : I did not intend the xattrs to have a prefix of ATTR_PREFIX here. In `get_attrs`, I use the prefix because the filter driver will be interested in all the user defined xattrs and they are prefixed with `ATTR_PREFIX` here. While setting the attrs, the prefix is added in the filter driver itself, so this may cause problems for other workflows. 
Where exactly are you seeing the error? If you set an attr using an attr_name, you must be able to retrieve the same via get_attr() using the same name. Does `set_attr` fail or `get_attr`?

Reply:
`set_attr` fails when I try to set the local weight for the block. Instead of adding the prefix in this method, I can add it to the "localWeight" attribute name when I am setting it in the filter. I can undo this commit as a result. Would that work?

Reply:
Do you mean calling set_attr, without adding the prefix to the "localWeight" attribute name, crashes?

Reply:
Yes, that is what I mean